Updated Project Members page with real data

jupiter007 opened the pull request:
## Description
The `Project Members` page was updated with real data. However, we need to go back and update it when we have the `isPrimaryInvestigator` data from the backend. I created a ticket for it here: https://github.com/CDLUC3/dmsp_frontend_prototype/issues/368

Tasks completed:
- Updated `Project Members` page to use real data
- Added new `ProjectContributors` query
- Added unit test
- Added translations

Fixes # ([360](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/360))

## Type of change
- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
Unit test and manual testing


## Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I updated the CHANGELOG.md and added documentation if necessary
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have completed manual or automated accessibility testing for my changes
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

fraserclark commented on app/[locale]/projects/[projectId]/members/page.tsx:
```diff
@@ -1,172 +1,209 @@
+
     // Handle editing member
-    window.location.href = '/projects/proj_2425/members/edit?memberid='+memberId;
+    router.push(`/projects/${projectId}/members/edit?memberid=${memberId?.toString()}`);
```
Long term probably we should create separate edit page like projects/[projectid]/member/[memberid]/edit

Semi related, but maybe it needs to go into a ticket; I noticed doing some stuff that having to build the routes is kind of painful - should we consider maybe a global helper function to help us with URL builder.

Like in the laravel world, there is a helper function that you can use normally it looks like 

- route('project.index')  => "/project"
- route('project.show', { projectId: 42 })  => "/project/42"
- route('project.dmp.question', { projectId: 42, dmpId: 7, questionId: 999 })  => "/project/42/dmp/7/q/999"

So that is a "named" route which expects 2 params which you name in the array basically

But open it being simpler like functions 

- getProjectIndex() => "/project"
- getProjectURL(123) => "/project/123"
- getProjectDetailsURL(123) => "/project/123/details"
- getProjectDmpQuestionURL(<project id>123, <plan id>555, <question id> 99) => "/project/123/dmp/555/q/99"

jupiter007 replied:
Thanks for your review @fraserclark. I agree that we should include a `memberId` param in the path, instead of tacking it on as a query param. I added a note to the `Edit Member Details` ticket [here](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/367). 

I'm getting worried about the length of the urls, but maybe we can shorten some of the directory names, like you did changing `question` to just `q`.

I'll create a ticket to look into a URL builder helper function. Thanks.

jupiter007 commented:
@fraserclark I updated the PR with the removal of "Update access" and "Share access" buttons, and updated the language above the "Share with people" link. If you approve of my updates, I'm going to go ahead and merge this ticket for now.

I have another ticket to go back and update the Project Members page once we have backend if for `Project Collaborators` and the `isPrimaryInvestigator` data.

![image](https://github.com/user-attachments/assets/d1c59d35-af94-4c01-b96d-abf37ec9984b)

briri commented:
The `IsPrimaryContact` flag is already available here: https://github.com/CDLUC3/dmsp_backend_prototype/blob/e8146f256854915495e01ab108fd1f5c749024c6/src/schemas/contributor.ts#L82

I think there is still confusion between **Principal** Investigator (which is just a title with regard to the tool) and **Primary** contact which is a designation that indicates that the individual will receive all requests for information about the DMP once it is published (has a DOI).

There can be any number (0..*) of Principal Investigators(PIs) on a project but there can be only one Primary contact.

jupiter007 commented:
> The `IsPrimaryContact` flag is already available here: https://github.com/CDLUC3/dmsp_backend_prototype/blob/e8146f256854915495e01ab108fd1f5c749024c6/src/schemas/contributor.ts#L82
> 
> I think there is still confusion between **Principal** Investigator (which is just a title with regard to the tool) and **Primary** contact which is a designation that indicates that the individual will receive all requests for information about the DMP once it is published (has a DOI).
> 
> There can be any number (0..*) of Principal Investigators(PIs) on a project but there can be only one Primary contact.

Thanks for the clarification @briri. I actually meant to remove the use of `isPrimaryContact` yesterday, since we no longer need it for the `Update access` and `Share access` buttons that we removed. I went ahead and removed them now.

briri reviewed (approved):
Sorry, I thought I had approved this one already. Looks like you just need to pull in the latest from the dev branch

jupiter007 commented:
> Sorry, I thought I had approved this one already. Looks like you just need to pull in the latest from the dev branch

Thanks @briri
